src/main/scala/edu/berkeley/ce/rockslicing/LoadBalancer.scala

```diff
@@ -0,0 +1,323 @@
+    val volumePerPart = rockVolume.volume/(numSeedJoints + 1)
+    val centroid = rockVolume.centroid
+    val centroidVolume = Block(centroid, rockVolume.updateFaces(centroid))
+    val seedJoints =
+      findProcessorJoints(Seq[Joint](), (centroidVolume.centerX, centroidVolume.centerY, centroidVolume.centerZ),
+                          (boundingBox._1, boundingBox._2, boundingBox._3),
+                          centroidVolume, volumePerPart)
+
+    if (seedJoints.length != numSeedJoints) {
+      throw new IllegalStateException("ERROR: LoadBalancer.generateSeedJoints unable to find optimal "+
+                                      "seed joints")
+    }
+    seedJoints
+  }
+
+  /**
+    * Finds the best set of processor joints to divide initial volume into approximately equal volumes
+    * @param joints List to prepend processor joints to during each recursion
+    * @param origin Origin which the joint is in reference to
+    * @param center Center of the joint plane - start with lower left corner of bounding box
+    * @param initialVolume Intial volume that needs to be sub-divided
+    * @param desiredVolume Desired volume for each piece cut by processor joints
+    * @return List of processor joints that will be used to seed initial RDD
+    */
+  @tailrec
+  private def findProcessorJoints(joints: Seq[Joint],
+                                  origin: (Double, Double, Double),
+                                  center: (Double, Double, Double), initialVolume: Block,
+                                  desiredVolume: Double):
+                                  Seq[Joint] = {
+    // Calculate bounding box
+    val vertices = initialVolume.findVertices.values.flatten
+    val x_max = (vertices.map{ case vertex => vertex._1}).max
+    val x_min = (vertices.map{ case vertex => vertex._1}).min
+    val y_max = (vertices.map{ case vertex => vertex._2}).max
+    val y_min = (vertices.map{ case vertex => vertex._2}).min
+    val z_max = (vertices.map{ case vertex => vertex._3}).max
+    val z_min = (vertices.map{ case vertex => vertex._3}).min
+    val boundingBox = (x_min, y_min, z_min, x_max, y_max, z_max)
+
+    // Calculate diagonal length
+    val x_diff = boundingBox._4 - boundingBox._1
+    val y_diff = boundingBox._5 - boundingBox._2
+    val z_diff = boundingBox._6 - boundingBox._3
+    val diagonalLength = math.sqrt(x_diff*x_diff + y_diff*y_diff + z_diff*z_diff)
+    val normal = linalg.normalize(DenseVector[Double](x_diff, y_diff, z_diff))
+
+    val joint = Joint((normal(0), normal(1), normal(2)), origin, center, phi = 0.0, cohesion = 0.0,
+                      shape = Nil, processorJoint = true)
+    val blocks = initialVolume.cut(joint)
+    val nonRedundantBlocks = blocks.map { case block @ Block(center, _) =>
+      Block(center, block.nonRedundantFaces)
+    }
+    val sortedBlocks = nonRedundantBlocks.sortWith{ (left, right) =>
+      centroidCompare(left.centroid, right.centroid) }
+    val tolerance = 0.05
```

Comment:
Sure, I think you have a good point. Would you mind making it a constant defined at the beginning of the class, i.e. write something like `val VOLUME_TOLERANCE = 0.5` ? That way we can easily keep track of it and make changes in the future.